centrallix/osdrivers/objdrv_xml.c

```diff
@@ -490,6 +490,12 @@ xml_internal_ReadDoc(pObject obj)
 	    /** parse the document **/
 	    ptr=(char*)malloc(XML_BLOCK_SIZE);
 	    ctxt=xmlCreatePushParserCtxt(NULL,NULL,NULL,0,"unknown");
+	    /** change the encoding if needed*/
+	    ObjData contentCharset;
+	    objGetAttrValue(obj->Prev, "content_charset", DATA_T_STRING, &contentCharset);
+	    if(contentCharset.String != NULL)
```

Comment:
We need to also check the return value of objGetAttrValue().  If it returns -1 then it failed to get the property, in which case contentCharset.String is unchanged and holds whatever uninitialized value was in it when it was declared.  A return value of -1 should be handled the same way a NULL value is handled in this case.

Reply:
I committed the requested changes. Because the XML cache bug still exists in this branch testing was not as thorough as I would have liked, but this change should not have been able to change much anyway